components/tidb_query_datatype/src/def/field_type.rs

```diff
@@ -98,15 +98,27 @@ impl From<FieldTypeTp> for ColumnInfo {
 /// Valid values of `tipb::FieldType::collate` and
 /// `tipb::ColumnInfo::collation`.
 ///
-/// The default value is `Utf8Bin`. See `charset.go` in pingcap/parser.
-#[derive(PartialEq, Debug, Clone, Copy, FromPrimitive, ToPrimitive)]
+/// Legacy Utf8Bin collator (was the default) does not pad. For compatibility,
+/// all new collation with padding behavior is negative.
+#[derive(PartialEq, Debug, Clone, Copy)]
 #[repr(i32)]
 pub enum Collation {
-    Utf8GeneralCi = 33,
```

Comment:
Maybe keep the legacy ones so we know where the magic numbers in `from_i32` come from?

Reply:
Utf8GeneralCi is not published in any version yet and 33 was undefined before., so it's a new collation id.

Reply:
But it's always strange to see the numbers like `-33`, `-46` in `from_i32`. Can we just not support them?

Reply:
why is it strange to see `-33`? it is valid collation id.

Reply:
all collation id in new cluster sent by tidb is negative.

Reply:
Doc added